massa-models/src/storage.rs

```diff
@@ -0,0 +1,70 @@
+use crate::prehash::Map;
+use crate::{Block, BlockId, Endorsement, EndorsementId, Operation, OperationId};
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+pub struct StoredBlock {
+    pub block: Block,
+    pub serialized: Vec<u8>,
+    pub serialized_header: Option<Vec<u8>>,
+}
+
+#[derive(Clone, Default)]
+pub struct Storage {
+    blocks: Arc<RwLock<Map<BlockId, Arc<RwLock<StoredBlock>>>>>,
+    operations: Arc<RwLock<Map<OperationId, Arc<RwLock<Operation>>>>>,
+    endorsements: Arc<RwLock<Map<EndorsementId, Arc<RwLock<Endorsement>>>>>,
+}
+
+impl Storage {
```

Comment:
I would love to rather have a something like a:
```rust
impl Storage for Block {
    pub fn store(&self, id: BlockId, item: Block, serialized: Vec<u8>) { todo!() }
    pub fn retrieve_block(&self, id: &BlockId) -> Option<Arc<RwLock<StoredBlock>>> { todo!() }
}
```
But this completely be part of a follow-up :slightly_smiling_face:!